smart-contracts/contracts/cl-vault/src/vault/swap.rs

```diff
@@ -29,53 +28,45 @@ pub struct SwapParams {
 pub fn execute_swap_non_vault_funds(
     deps: DepsMut,
     env: Env,
-    info: MessageInfo,
+    _info: MessageInfo,
     swap_operations: Vec<SwapOperation>,
 ) -> Result<Response, ContractError> {
-    // validate auto compound admin as the purpose of swaps are mainly around autocompound non-vault assets into assets that can be actually compounded.
-    assert_range_admin(deps.storage, &info.sender)?;
-
     let vault_config = VAULT_CONFIG.load(deps.storage)?;
     let pool_config = POOL_CONFIG.load(deps.storage)?;
 
     if swap_operations.is_empty() {
         return Err(ContractError::EmptySwapOperations {});
     }
 
-    let mut swap_msgs: Vec<CosmosMsg> = vec![];
+    let contract_address = env.clone().contract.address;
+    let mut swap_msgs = Vec::new();
 
     for swap_operation in swap_operations {
-        let token_in_denom = swap_operation.token_in_denom.clone();
-        let pool_token_0 = pool_config.token0.clone();
-        let pool_token_1 = pool_config.token1.clone();
+        let token_in_denom = &swap_operation.token_in_denom;
 
         // Assert that no BASE_DENOM or QUOTE_DENOM is trying to be swapped as token_in
-        if token_in_denom == pool_token_0 || token_in_denom == pool_token_1 {
+        if token_in_denom == &pool_config.token0 || token_in_denom == &pool_config.token1 {
             return Err(ContractError::InvalidSwapAssets {});
         }
 
-        // Throw an Error if contract balance for the wanted denom is 0
+        // Get contract balance for the input token
         let balance_in_contract = deps
             .querier
-            .query_balance(
-                env.clone().contract.address,
-                swap_operation.clone().token_in_denom,
-            )?
+            .query_balance(contract_address.clone(), token_in_denom.clone())?
             .amount;
 
-        // TODO_FUTURE: This could be a <= condition against a threshold value mayube in dollars to avoid dust swaps
-        if balance_in_contract == Uint128::zero() {
-            // TODO: Use InsufficientFundsForSwap instead, this has been removed after STRATEGIST_REWARDS state eval removal
-            return Err(ContractError::InsufficientFunds {});
+        if balance_in_contract.is_zero() {
+            return Err(ContractError::InsufficientFundsForSwap {
+                balance: balance_in_contract,
+                needed: Uint128::new(1),
+            });
         }
 
-        // TODO_FUTURE: We could be swapping into the actual vault balance so we could prepend_swap() the autocompound entrypoint.
+        // Split balance into two parts
```

Comment:
this comment does not provide more information than the following code, plz remove